subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -1544,74 +1545,156 @@ static int cmd_wifi_twt_setup_quick(const struct shell *sh, size_t argc,
 	return 0;
 }
 
-static int cmd_wifi_twt_setup(const struct shell *sh, size_t argc,
-			      char *argv[])
+static int twt_args_to_params(const struct shell *sh, size_t argc, char *argv[],
```

Comment:
can you please do the getopt conversion in a separate commit

Reply:
There is no plan to make the getopt conversion in a separate commit.
Actually one of the sub-commit almost all related with this twt_args_to_params, do you mean to make the fix of twt quick setup and twt tear down in  separate commit?

Reply:
basically 1 commit for getopt conversion and another commit for extending TWT params

Reply:
> Add parameters for twt setup and enhance the parse parameters code

`and` implies that these can split in two commits, can you please do that.

Reply:
There will be not clearly if separate getopt conversion  into a commit.
So I make the PR in 4 separated commits base on the commits function.

Reply:
Yes, of course, this is reasonable, separated as comment, also make quick setup and teardown in two separated commits to make code logic more clearly